en/guide/create_message.md

```diff
@@ -0,0 +1,246 @@
+      <field type="int16_t" name="current_battery" units="cA">Battery current, -1: autopilot does not measure the current</field>
+      <field type="int32_t" name="current_consumed" units="mAh">Consumed charge, -1: autopilot does not provide consumption estimate</field>
+      <field type="int32_t" name="energy_consumed" units="hJ">Consumed energy, -1: autopilot does not provide energy consumption estimate</field>
+      <field type="int8_t" name="battery_remaining" units="%">Remaining battery energy. Values: [0-100], -1: autopilot does not estimate the remaining battery.</field>
+      <extensions/>
+      <field type="int32_t" name="time_remaining" units="s">Remaining battery time, 0: autopilot does not provide remaining battery time estimate</field>
+      <field type="uint8_t" name="charge_state" enum="MAV_BATTERY_CHARGE_STATE">State for extent of discharge, provided by autopilot for warning or external reactions</field>
+    </message>
+```
+    
+
+The main message tags/fields are:
+
+- `message`: Each message is encapsulated by `message` tags, with the following attributes
+  - `id`: The id attribute is the unique index number of this message (in this case 147). 
+    - For MAVLink 1:
+      - Valid numbers range from 0 to 255.
+      - The ids 0-149 and 230-255 are reserved for *common.xml*. Dialects can use 150-229 (?240) for custom messages (provided these are not used by other included dialects). 
+    - For MAVLink 2:
+      - Valid numbers range from 0 to 16777215.
+      - All numbers below 255 should be considered reserved unless messages are also intended for MAVLink 1. 
+        > **Note** IDs are precious in MAVLink 1!
+  - `name`: The name attribute provides a human readable form for the message (ie "BATTERY_STATUS"). It is used for naming helper functions in generated libraries, but is not sent over the wire.
+- `description` (optional): Human readable description of message, shown in user interfaces and in code comments. 
+  This should contain all information (and hyperlinks) to fully understand the message.
+- `field`: Encodes one field of the message. The field value is its name/text string used in GUI documentation (but not sent over the wire).
+  - `type`: Similar to a field in a C struct - the size of the data required to store/represent the data type.
+    - Fields can be signed/unsigned integers of size 8, 16, 23, 64 bits (`{u)int8_t`, `(u)int16_t`, `(u)int32_t`, `(u)int64_t`), single/double precision IEEE754 floating point numbers.They can also be arrays of the other types - e.g. `uint16_t[10]`. 
+  - `name`: Name of the field (used in code).
+  - `enum`: Name of an enum defining possible values of the field (e.g. `MAV_BATTERY_CHARGE_STATE`).
+  - `units`: The units for fields that take numeric values (not enums). These are defined in the [schema](https://github.com/ArduPilot/pymavlink/blob/master/generator/mavschema.xsd) (search on *name="SI_Unit"*)
```

Comment:
might be a good idea to also document `print_format` and `display` attributes.
There is also an attribute called `default`that seams to not be used anywhere. That one should maybe removed from the spec. @LorenzMeier what do you think?